unit_tests/test_name_resolving.py

```diff
@@ -0,0 +1,68 @@
+import struct
+
+from pytest_bdd import given, parsers, scenarios, then, when
+
+from boofuzz import *
+from boofuzz.mutation_context import MutationContext
+
+CONVERTERS = {
+    "block_name": str,
+}
+
+scenarios("test_name_resolving.feature", example_converters=CONVERTERS)
+
+
+@given("A flat scenario with block_name <block_name>")
+def a_flat_scenario(context, block_name):
+    global blocks
+    blocks.REQUESTS = {}
+    s_initialize("test_name_resolving")
+    s_static(b"\xff\xff", name="2_static_bytes")
+    s_size(block_name, length=1)
+    s_static(b"\xff", name="1_static_byte")
+    context.req = s_get("test_name_resolving")
+
+
+@given("A 1 deep scenario with block_name <block_name>")
+def a_1_deep_scenario(context, block_name):
+    global blocks
+    blocks.REQUESTS = {}
+    s_initialize("test_name_resolving")
+    s_static(b"\xff", name="header")
+    s_size(block_name, length=1)
+    if s_block_start("1_deep"):
```

Comment:
Using `with s_block()` is preferred over `if s_block_start`
https://boofuzz.readthedocs.io/en/stable/user/static-protocol-definition.html#boofuzz.s_block_start

We should perhaps make this more clear in the docs and mark `s_block_start` as deprecated.

Reply:
I thought `with` was not python 2 compatible, but just tested it and it works. Will fix.